controllers/grid/files/form/ManageSubmissionFilesForm.inc.php

```diff
@@ -80,32 +80,39 @@ function execute($args, $request, $stageSubmissionFiles, $fileStage) {
 			// Get the viewable flag value.
 			$isViewable = in_array(
 				$submissionFile->getFileId(),
-				$selectedFiles);
+				$selectedFiles
+			);
 
 			// If this is a submission file that belongs to the current stage id...
 			if (array_key_exists($submissionFile->getFileId(), $stageSubmissionFiles)) {
 				// ...update the "viewable" flag accordingly.
 				$submissionFile->setViewable($isViewable);
-			} else {
-				// If the viewable flag is set to true...
-				if ($isViewable) {
-					// Make a copy of the file to the current file stage.
-					import('lib.pkp.classes.file.SubmissionFileManager');
-					$context = $request->getContext();
-					$submissionFileManager = new SubmissionFileManager($context->getId(), $submissionFile->getSubmissionId());
-					// Split the file into file id and file revision.
-					$fileId = $submissionFile->getFileId();
-					$revision = $submissionFile->getRevision();
-					list($newFileId, $newRevision) = $submissionFileManager->copyFileToFileStage($fileId, $revision, $fileStage, null, true);
-					if ($fileStage == SUBMISSION_FILE_REVIEW_FILE) {
-						$submissionFileDao->assignRevisionToReviewRound($newFileId, $newRevision, $this->getReviewRound());
-					}
-					$submissionFile = $submissionFileDao->getRevision($newFileId, $newRevision);
-				}
+				$submissionFileDao->updateObject($submissionFile);
+			} elseif ($isViewable) {
+				// Import a file from a previous stage.
+				$context = $request->getContext();
+				if ($isViewable) $submissionFile = $this->importFile($context, $submissionFile, $fileStage);
 			}
-			$submissionFileDao->updateObject($submissionFile);
 		}
 	}
+
+	/**
+	 * Make a copy of the file to the current file stage.
+	 * @param $context Context
+	 * @param $submissionFile SubmissionFile
+	 * @param $fileStage int SUBMISSION_FILE_...
+	 * @return SubmissionFile Resultant new submission file
+	 */
+	protected function importFile($context, $submissionFile, $fileStage) {
+		$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO');
```

Comment:
admit, you like Crockford's approach on variable initialization... I would initialize it closer to where it's used :P